vmm/src/cpu.rs

```diff
@@ -321,23 +321,39 @@ impl Vcpu {
             Ok(run) => match run {
                 #[cfg(target_arch = "x86_64")]
                 VmExit::IoIn(addr, data) => {
-                    self.io_bus.read(u64::from(addr), data);
+                    if let Err(e) = self.io_bus.read(u64::from(addr), data) {
+                        if let devices::BusError::MissingAddressRange = e {
```

Comment:
```suggestion
                        if devices::BusError::MissingAddressRange == e {
```
Otherwise, you're just assigning a value with `let`, aren't you?

Reply:
```
Compiling vmm v0.1.0 (/home/samuel/devlp/hypervisor/cloud-hypervisor/vmm)
error[E0369]: binary operation `==` cannot be applied to type `vm::devices::BusError`
   --> vmm/src/cpu.rs:325:67
    |
325 |                         if devices::BusError::MissingAddressRange == e {
    |                            -------------------------------------- ^^ - vm::devices::BusError
    |                            |
    |                            vm::devices::BusError

error: aborting due to previous error

```

Reply:
I guess that's because the error does not derive `PartialEq`.

Reply:
Just for the sake of clarifying, I've tried deriving `PartialEq` for `BusError` and it works as expected.
It took me some time to understand this syntax but let's keep your way as it is more idiomatic of Rust :)